### PR TITLE
Fixes parser restart bug with UDFs

### PR DESCRIPTION
**Pull request by VamsiModem**
Parser wasn't restarting when a return type is missing in UDF, this PR fixes that

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by Shpakh-MSFT** [comment]
>         [InlineData("a = Abs(1.2);\nAdd(a: Number, b: Number):Number = a + b;\nb = Add(1, 2);", 2, 1, 0)]

Can you adda test for following?

a = Abs(1.2);\nAdd(a: Number, b: Number):/*  Number */Number

---
Refers to: src/tests/Microsoft.PowerFx.Core.Tests/UserDefinedFunctionTests.cs:195 in 514b4b2. [](commit_id = 514b4b2472f98c47c5ffb453b7882dadc2b4984e, deletion_comment = False)

File: src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs [hunk 1]
```diff
@@ -301,7 +301,10 @@ private ParseUserDefinitionResult ParseUDFsAndNamedFormulas(string script, Parse
                     {
                         CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingReturnType);
                         udfs.Add(new UDF(thisIdentifier.As<IdentToken>(), colonToken, returnType: null, new HashSet<UDFArg>(args), body: null, _hasSemicolon, parserOptions.NumberIsFloat, isValid: false));
-                        break;
+
+                        // Return type not found, move to next user definition
+                        MoveToNextUserDefinition();
+                        continue;
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs:307]: Why do we need this for return type?

Comment (VamsiModem) [reply src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs:307]: When a return type is not found it's still considered a parser error and we must move to the next user definition.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by Shpakh-MSFT — APPROVED** [review]
:shipit:

**Pull request merged**